### ISSUE TITLE
question (not issue) : event management

### ISSUE THREAD

**Issue by brucou** [TargetProcess/tauCharts#529]
Hi there, first of all congratulations for a really nice library with good-looking defaults.

I am evaluating several graph libraries and there is one thing that is mandatory for me to move forward with a given library, and I could not get that information from the API documentation. 

I need to be able to capture (left and right) click on specific points of the graph, and associate that to an event handler which would receive in parameters the specific points which were clicked on. Then the clicked-on point might have to change its colour (to indicate selection).

Is there a way to do this with tauCharts??

Thanks!

**Comment by vladminsky** [comment]
Hi @brucou 

Thank you for a kind words.

Have you seen an examples on https://api.taucharts.com/plugins/create-plugin.html page?
Does it look similar to what you need?

Actually taucharts doesn't support right-click handling at the moment. But left-click event expose data of the clicked point. The look and feel of highlighted point is controlled using special CSS class which can be customized.

**Comment by brucou** [comment]
Thanks for a quick answer. Alright, I should be able to do without right-clicking. 
Yes I saw the examples previously but I am a bit dumbfounded here. None of those examples feature an event handler. Maybe I should look at `tooltip` plugin code? What I miss is the description of the parameters that event handler take, and where does that event handler appear. Is it in `onSpecReady: function(chart, specRef)` ?? Going to have a look now.

It would be great to have some, even succint description, of the API for API-users. The existing documentation is pretty good for end-users though.

**Comment by vladminsky** [comment]
@brucou you right. I have to reconsider documentation and some technical decisions.
Indeed the project api is not ready for public plugins development at the moment.

**Comment by brucou** [comment]
That is all right. Everyhing will come in its due time. As mentioned, one of the important part of the plugin features will be bi-directional event communication with outside environment, (read and write) access to data enclosed in the graph (and more generally the interesting parts of the graph state), and handling rerendering of the graph if its inner data changes due to the plugin operation.

In my use case (tracing tool, analyzing incoming log messages), my data is streamed so I update the graph for every (more or less) incoming piece of data. I also seek to allow brushing facilities, so I am interested in the user being able to select a rectangle of the graph and get the data point selected for some external tool to process and display reports.

To that purpose, for instance `http://www.chartjs.org/docs/latest/developers/api.html` i.e. `chartjs` exposes an `.getElementAtEvent(e)` and `.getElementsAtEvent(e)` to get the selected (brushed data), so one probably can declare an event handler on the chart css selector, and get the selected data from there. There also is a `render` method (http://www.chartjs.org/docs/latest/developers/updates.html) which updates the graph if the data has changed.

I am not seeking to recommend to copy this approach at all, I am just giving you this information to feed your reflection and technical decisions. As is, `tauCharts` is already an amazing piece of work.

**Comment by brucou** [comment]
@vladminsky Hi Vladimir. Was wondering if there any progress with this issue. If not, I am happy to close it.

**Comment by Mavrin** [comment]
Hi @brucou.
This issue is still relevant. Of course you can close issue. It is up to you. 
Thank you.

**Comment by brucou** [comment]
Thanks @Mavrin for maintaining this superb piece of software. ALright I will leave it open still.
